addon/utils/resource-helper.js

```diff
@@ -0,0 +1,38 @@
+import Ember from 'ember';
+import config from 'ember-get-config';
+const {get, $} = Ember;
+
+export default {
+  loadJavascript(url) {
+    let async = get(config, 'ember-cli-bundle-loader.asyncScriptExecution');
+    if (async === undefined) {
+      // For backwards compatibility we use async true
+      async = true;
+    }
+
+    var scriptElement = $("<script>").prop({src: url, async });
+    let promise = new Ember.RSVP.Promise((resolve, reject)=>{
+      scriptElement.one('load', ()=> Ember.run(null, resolve));
+      scriptElement.one('error', (evt)=> Ember.run(null, reject, evt));
```

Comment:
If there is an error, can you remove the element? 
At the moment, if there is a temp  error loading the script, retrying the transition will attempt to load the file again. With the code added to check `isJavascriptLoaded`. One option to assume we maintain that retry behavior for errors is to remove the element.